webui/react/src/pages/FlatRuns/FlatRuns.settings.ts

```diff
@@ -0,0 +1,81 @@
+import * as t from 'io-ts';
+
+import { INIT_FORMSET } from 'components/FilterForm/components/FilterFormStore';
+import { SettingsConfig } from 'hooks/useSettings';
+import { DEFAULT_SELECTION, SelectionType } from 'pages/F_ExpList/F_ExperimentList.settings';
+
+import { defaultColumnWidths, defaultRunColumns } from './columns';
+
+// have to intersect with an empty object bc of settings store type issue
+export const FlatRunsSettings = t.intersection([
+  t.type({}),
+  t.type({
+    columns: t.array(t.string),
+    columnWidths: t.record(t.string, t.number),
+    compare: t.boolean,
+    filterset: t.string, // save FilterFormSet as string
+    pageLimit: t.number,
+    pinnedColumnsCount: t.number,
+    selection: SelectionType,
+    sortString: t.string,
+  }),
+]);
+export type FlatRunsSettings = t.TypeOf<typeof FlatRunsSettings>;
+
+export const ProjectUrlSettings = t.partial({
+  compare: t.boolean,
+  page: t.number,
+});
+
+export const settingsConfigForProject = (id: number): SettingsConfig<FlatRunsSettings> => ({
```

Comment:
can we just use direct settings here? in particular, because the selection state is stored in the settings store but the update happens outside of the event loop, it can appear laggy.

Reply:
What do you mean by direct settings?

Reply:
instead of using `useSettings` directly interface with the settings store.